src/index.js

```diff
@@ -1,100 +1,95 @@
-import 'katex/contrib/mhchem/mhchem';
-import katex from 'katex';
-
-let options = {
-  throwOnError: false,
-  displayMode: false
-};
-let blockOptions = {
-  throwOnError: false,
-  displayMode: true
-};
-
-const magicEscapedDollar = 'c194a9eb';
-const magicEscapedDollarRegex = /c194a9eb/g;
-const magicBacktickInCodeTag = 'c194a9ec';
-const magicBacktickInCodeTagRegex = /c194a9ec/g;
-const magicBacktickInDollars = 'c194a9ed';
-const magicBacktickInDollarsRegex = /c194a9ed/g;
-const magicEscapedBacktick = 'c194a9ee';
-const magicEscapedBacktickRegex = /c194a9ee/g;
-const magicDollarInBacktick = 'c194a9ef';
-const magicDollarInBacktickRegex = /c194a9ef/g;
-
-const preMathInlineOpen = 'c194a9eg<!-- begin-inline-katex';
-const preMathInlineClose = 'end-inline-katex-->';
-const preMathInlineRegex = /c194a9eg<!-- begin-inline-katex([\s\S]*?)end-inline-katex-->/g;
+// import katex from "katex";
+// import marked from "marked";
 
+(function () {
+  let oldMarkdown = window.$docsify.markdown;
+  let newMarked = marked;
+  let newM = JSON.stringify(newMarked);
+  function newMarkdown(originMarked, originRenderer) {
+    // let newMarked = marked; // in docsify.js: window.$marked = marked
+    function isFn(obj) {
+      return typeof obj === "function";
+    }
+    const mathExtension = {
+      name: "math",
+      level: "inline",
+      start(src) {
+        let index = src.match(/\$/)?.index;
+        return index;
+      },
+      tokenizer(src, tokens) {
+        const blockRule = /^\$\$((\\.|[^\$\\])+)\$\$/;
+        const inlineRule = /^\$((\\.|[^\$\\])+)\$/;
+        let match;
+        if ((match = blockRule.exec(src))) {
+          return {
+            type: "math",
+            raw: match[0],
+            text: match[1].trim(),
+            mathLevel: "block",
+          };
+        } else if ((match = inlineRule.exec(src))) {
+          return {
+            type: "math",
+            raw: match[0],
+            text: match[1].trim(),
+            mathLevel: "inline",
+          };
+        }
+      },
+      renderer(token) {
+        if (token.mathLevel === "block") {
+          return (
+            '<div class="math">' +
```

Comment:
Why add an extra div with class `math`?

Reply:
I am unfamiliar with Html, so I don't know what the element div means. To emphasize mathematical expressions as blocks, I just decorate them with div.

Reply:
I think we do not need this div here

Reply:
I have deleted it